website/src/pages/Visualization.vue

```diff
@@ -0,0 +1,708 @@
+<script setup>
+import {
+  onMounted,
+  onUnmounted,
+  ref,
+  computed,
+  provide,
+  h,
+  render,
+  shallowRef,
+  watch
+} from 'vue'
+import { get, set } from '@vueuse/core'
+import { Map, NavigationControl, Popup, LngLat } from 'maplibre-gl'
+import { MapboxOverlay as DeckOverlay } from '@deck.gl/mapbox'
+import { GeoJsonLayer } from '@deck.gl/layers'
+import bboxPolygon from '@turf/bbox'
+import {
+  isMapboxURL,
+  transformMapboxUrl
+} from 'maplibregl-mapbox-request-transformer'
+
+import VehiclePopup from '../components/VehiclePopup.vue'
+import StationPopup from '../components/StationPopup.vue'
+import GeofencingZonePopup from '../components/GeofencingZonePopup.vue'
+
+const result = shallowRef([])
+const summary = ref({})
+const isLoading = ref(false)
+const url = ref('')
+const map = ref()
+const container = ref()
+const selected = ref(['stations', 'vehicles'])
+
+let mapInstance
+let vehiclesOverlay
+let stationsOverlay
+let geofencingOverlay
+
+const MAPBOX_KEY =
+  'pk.eyJ1IjoicGllcnJpY2twIiwiYSI6ImNsaWluZjl6bDF3emQzZW9iZHZ2NDd5MDMifQ.FCTTnlXknFjxmw9NLemYsQ'
```

Comment:
Thank you @PierrickP for this amazing contribution!

We set up an environment variable in Netlify to avoid exposing the Mapbox key in the code.

We created three tokens under MobilityData Mapbox account and added to Netlify `VITE_MAPBOX_API_KEY` per context:

- PROD
- STAGING - branch the preview deployment
- LOCAL - local netlify cli

Please let us know if you agree with this approach. Thank you!

```suggestion
const MAPBOX_KEY = import.meta.env.VITE_MAPBOX_API_KEY
```

cc @davidgamez

Reply:
That's an excellent improvement. Thanks for that. I've stored the Tokens in 1Password, in the Public transit vault.

Reply:
:+1: i made the changes